stress/basic.go

```diff
@@ -0,0 +1,539 @@
+	var buf bytes.Buffer
+	var wg sync.WaitGroup
+	counter := NewConcurrencyLimiter(c.Concurrency)
+
+	interval, err := time.ParseDuration(c.BatchInterval)
+	if err != nil {
+		// actually do error
+	}
+
+	ctr := 0
+
+	for p := range ps {
+		b := p.Line()
+		ctr++
+
+		buf.Write(b)
+		buf.Write([]byte("\n"))
+
+		if ctr%c.BatchSize == 0 && ctr != 0 {
+			b := buf.Bytes()
+
+			b = b[0 : len(b)-2]
+
+			wg.Add(1)
+			counter.Increment()
+			go func(byt []byte) {
+
+				rs := c.send(byt)
+				time.Sleep(interval)
+
+				counter.Decrement()
+				r <- rs
+				wg.Done()
+			}(b)
+
+			var temp bytes.Buffer
+			buf = temp
+		}
+
+	}
+
+	wg.Wait()
+}
+
+// post sends a post request with a payload of points
+func post(url string, datatype string, data io.Reader) (*http.Response, error) {
+
+	resp, err := http.Post(url, datatype, data)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusNoContent && resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf(string(body))
+	}
+
+	return resp, nil
+}
+
+// Send calls post and returns a response
+func (c *BasicClient) send(b []byte) response {
+	instanceURL := fmt.Sprintf("http://%v/write?db=%v&precision=%v", c.Address, c.Database, c.Precision)
+	t := NewTimer()
+
+	t.StartTimer()
+	resp, err := post(instanceURL, "application/x-www-form-urlencoded", bytes.NewBuffer(b))
+	t.StopTimer()
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	r := response{
+		Resp:  resp,
+		Time:  time.Now(),
+		Timer: t,
+	}
+
+	return r
+}
+
+// BasicQuery implements the QueryGenerator interface
+type BasicQuery struct {
+	Template   Query `toml:"template"`
+	QueryCount int   `toml:"query_count"`
+	time       time.Time
+}
+
+// QueryGenerate returns a Query channel
+func (q *BasicQuery) QueryGenerate() <-chan Query {
+	c := make(chan Query, 0)
+
+	go func(chan Query) {
+		defer close(c)
+
+		for i := 0; i < q.QueryCount; i++ {
+			time.Sleep(10 * time.Millisecond)
+			c <- Query(fmt.Sprintf(string(q.Template), i))
+		}
+
+	}(c)
+
+	return c
+}
+
+// SetTime sets the internal state of time
+func (q *BasicQuery) SetTime(t time.Time) {
+	q.time = t
+
+	return
+}
+
+// BasicQueryClient implements the QueryClient interface
+type BasicQueryClient struct {
+	Address       string `toml:"address"`
+	Database      string `toml:"database"`
+	QueryInterval string `toml:"query_interval"`
+	Concurrency   int    `toml:"concurrency"`
+	client        client.Client
+}
+
+// Init initializes the InfluxDB client
+func (b *BasicQueryClient) Init() {
+	u, _ := url.Parse(fmt.Sprintf("http://%v", b.Address))
+	cl := client.NewClient(client.Config{
+		URL: u,
+	})
+
+	b.client = cl
+}
+
+// Query runs the query
+func (b *BasicQueryClient) Query(cmd Query, ts time.Time) response {
+	q := client.Query{
+		Command:  string(cmd),
+		Database: b.Database,
+	}
+
+	t := NewTimer()
+
+	t.StartTimer()
```

Comment:
I actually don't need line 418. `NewTimer` will start the timer for me.

I had previously done
```
start := time.Now()
taken := time.Since(start)
```
but didn't like introducing the extra `start` variable, when all I care about is taken.

I'm happy to change it back though if you think its more better.